common/app/routes/challenges/redux/reducer.js

```diff
@@ -74,6 +75,7 @@ const mainReducer = handleActions(
       // used mainly to find code storage
       legacyKey: challenge.name,
       challenge: challenge.dashedName,
+      blockType: challenge.block,
```

Comment:
I had to reassign the `block` from `chalengeSelector`, trying to reassign `entities.block` still resulted in a name clash.

```js
const mapStateToProps = createSelector(
  challengeSelector,
  state => state.challengesApp.challenge,
  state => state.challengesApp.superBlocks,
  state => state.entities.block,
  (
    {
      challenge: { title = 'Learn to Code' } = {},
      viewType,
      block: blockIndex
    },
    challenge,
    superBlocks = [],
    block
  ) => ({
    title,
    viewType,
    challengeBlock: block[blockIndex], // block[blockIndex].title results in a loopback error,
// Cannot read property 'title' of undefined
    areChallengesLoaded: superBlocks.length > 0
  })
);
```
```js
// Show.jsx
componentDidMount() {
    const { areChallengesLoaded, title, challengeBlock } = this.props;
    const blockTitle = challengeBlock ? challengeBlock.title : null;
    if (!areChallengesLoaded) {
      this.props.fetchChallenges();
    }
    this.props.updateTitle({ blockTitle, title });
  }
```
```js
// Side-Panel.jsx render()
    const blockTitle = challengeBlock ? challengeBlock.title : null;
    const titleString = challengeTitle({ blockTitle, title });
    console.log( titleString);
```
The console.log shows the correct values being returned from 'challengeTitle()` but the challenge title remains as just `title`, no `blockTitle`. The same is seen for `updateTtile()` in `Show.jsx`. 

We could simplify the code by adding the `name || title` property to `state.challengesApp`. Using `entities` puts a delay in the render of the correct title due to the async nature of it, the same is seen in `Map`·